app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,53 @@
+class MiqCockpitWsWorker::Authenticator
```

Comment:
Perhaps this should be a module that gets included in the `MiqCockpitWsWorker` model since I belive these methods rely on `self` being an instance of that worker.